### ISSUE TITLE
PlotWidget tick labels are all the same for very small data range

### ISSUE THREAD

**Issue by t20100** [silx-kit/silx#3997]
To reproduce:
```python
from silx import sx
sx.plot((0, 1), (0, 1e-9))
```

Related to https://gitlab.esrf.fr/bliss/bliss/-/issues/4059

**Comment by vallsv** [comment]
Ideally if a scientific mode + offset can be enabled automatically when the displayed range is weird, it would be nice.

**Comment by t20100** [comment]
This is actually the case for matplolib:

- [offset is only used if it saves at least 4 digits by default](https://github.com/matplotlib/matplotlib/blob/ca946ac647e4a9f3c2b7a9ff27c11cfa2387b20d/lib/matplotlib/mpl-data/matplotlibrc#L402-L405)
- [trigger of using scientific notation can be configured](https://matplotlib.org/stable/api/ticker_api.html#matplotlib.ticker.ScalarFormatter.set_powerlimits), default is for absolute value < 10^-5 and > 10^6 

Now, when both are combined like for `sx.plot((0, 1), (10, 10+1e-9))`, it is hard to parse visually:
<img width="131" alt="Screen Shot 2023-12-15 at 11 03 05" src="https://github.com/silx-kit/silx/assets/9449698/42d9abac-f1df-4367-a8d8-b0cef0ff1555"> 

But there is a `useMathText` mode that makes it better:
<img width="146" alt="Screen Shot 2023-12-15 at 11 06 04" src="https://github.com/silx-kit/silx/assets/9449698/6f662fe3-0a77-475c-8c64-1c0090ee6e19">

I consider switching to: enable useOffset + scientific notation (with default "power limits") + useMathText for matplotlib versions that supports all of this and to leave everything disabled for older version.
What do you think?
